### PR TITLE
Task 8. Cart Api. Done

### PR DESCRIPTION
**Pull request by JenkaBY**
## Integration BE with FE has been done

I'm not FE developer. But most work in this task requires JS/TS expertise therefore there's a delay in BE integration task completion. 
For testing purpose you can use [the deployed FE application](https://d1hxiyo0kk0n80.cloudfront.net/). It has several bugs on the main page but main flow with ordering and checkout can be tested.
The application is used authorization and requires logged in user. There's no special page for authorization, instead you need to set 2 variables into localStorage:
```js
localStorage.setItem('username','anyusername');
localStorage.setItem('password','anypassword');
```
By default, the 'default' is used for both variables.

To check orders we can use [GET orders](https://fzpbwjly4d.execute-api.eu-north-1.amazonaws.com/prod/api/order). It's available without authorization.

### Stack:
- java 17
- gradle
- aws cdk v2
- angular as FE
- nest as BE for Cart Api 

#### Useful URL: 
- The working application is available [here](https://d1hxiyo0kk0n80.cloudfront.net/)
- CartApi is avaliable here - https://fzpbwjly4d.execute-api.eu-north-1.amazonaws.com/prod/ . Swagger or other links will be provided later.
#### Main tasks:
- [x] task-8.1 You should provide screenshot from CloudWatch with successful initialization of Typeorm module as well as the link for the [API Gateway](https://fzpbwjly4d.execute-api.eu-north-1.amazonaws.com/prod/).
image
Db Instance has been created via [CDK](https://github.com/JenkaBY/rss-aws-cloud-developer/pull/7/files#diff-42511d8cc4368eb0bee257b0624d834030e4c7d7929813ae682cb39fad85f56aR119-R176) 
- [x] task-8.2 As for task's wording - sql script should be provided. Additionally attach the screenshot from AWS Console with created DB. 
Db initialization was done by running migrations via java liqiubase. The migrations are running right after DbInstance will be created. It's achieved by [TriggerFunction](https://github.com/JenkaBY/rss-aws-cloud-developer/pull/7/files#diff-42511d8cc4368eb0bee257b0624d834030e4c7d7929813ae682cb39fad85f56aR68-R80). The source files are under the module **backend/cart-service/lambda-dbchangelog/** .The migrations files are:
- [carts and cart_items tables](https://github.com/JenkaBY/rss-aws-cloud-developer/pull/7/files#diff-1ddb42ce866d9a634a10d13659ad83b8c7c26d711a89787aba79480d992212e8)
- orders [table ](https://github.com/JenkaBY/rss-aws-cloud-developer/pull/7/files#diff-2d2e940e469f2704919151b0ae298aba8efac02afb56272e6758771c2290e12d)
- users [tables ](https://github.com/JenkaBY/rss-aws-cloud-developer/pull/7/files#diff-91eb2b405b59f9c754fa4d65229d2aa1b87418384c3e85e70b914a512b9742c1)
- [x] task-8.3 update the code in the appropriate files (app.module.ts and entities/ folder) to make it possible for application to connect to your database. There're several screenshots with proofs
image
Now you can test a connection via UI.

#### Additional ones - 4 points each:
- [x] Create orders table and integrated with it Order model: 
To check orders we can use [GET orders](https://fzpbwjly4d.execute-api.eu-north-1.amazonaws.com/prod/api/order). It's available without authorization.
```
curl https://fzpbwjly4d.execute-api.eu-north-1.amazonaws.com/prod/api/order
```
- [x] Create users table and integrate with it.
To test it, you need either postman or curl or another tools to send http request:
1. Send POST request : 
```
curl -X POST https://fzpbwjly4d.execute-api.eu-north-1.amazonaws.com/prod/api/auth/login  -d '{"username":"testFromReviewer","password":"testFromReviewer"}'
```
It'll respond with 
```json
{"statusCode":200,"message":"OK","data":{"token_type":"Basic","access_token":"dGVzdEZyb21SZXZpZXdlcjp0ZXN0RnJvbVJldmlld2Vy"}}
```
2. then you need to concatenate `${token_type} ${access_token}` and use it in the 'Authorization' header for almost each http request. Get order is not require authorization header.
- [x] Transaction based creation of checkout. There's a [proof ](https://github.com/JenkaBY/rss-aws-cloud-developer/pull/7/files#diff-af9ea9a0435fc33df5ffc4d1bbd0d8b31baf07b12d5bc0bf5b9cd3e142757011R25-R51)
- [x] Integrate Cart service with FE repository.

Estimation:
Score 100/100

### CODE REVIEW

File: backend/cart-service/deploy/src/main/java/by/jenka/rss/backend/cartservice/config/AwsConfig.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package by.jenka.rss.backend.cartservice.config;
+
+
+public class AwsConfig {
+
+    public static String DEFAULT_REGION = "eu-north-1";
+    public static String DEFAULT_ACCOUNT = "590184055641";
```
Comment (hazardsoft) [inline backend/cart-service/deploy/src/main/java/by/jenka/rss/backend/cartservice/config/AwsConfig.java:7]: IMHO should be taken from .env file and not hard coded.

File: backend/cart-service/lambda/src/cart/cart.controller.ts
Comment (hazardsoft) [inline backend/cart-service/lambda/src/cart/cart.controller.ts:1]: `cart-service/lambda/src` in this PR does not reflect changes, it looks like the repository was added from scratch, while it should contain changes only.
Could you please update PR in the way it contains Task 8 changes only?

File: backend/cart-service/lambda/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+async function bootstrap() {
+    const app = await NestFactory.create(AppModule);
+
+    app.enableCors({
+        origin: (req, callback) => callback(null, true),
+    });
+    app.use(helmet());
+
+    await app.listen(port);
```
Comment (hazardsoft) [inline backend/cart-service/lambda/src/main.ts]: Nest.js application should NOT listen if it's used in Lambda.

**Review by hazardsoft — CHANGES_REQUESTED** [review]
1. Didn’t find a place how `node_modules` are placed into lambda. Could you please point me?
2. Didn’t get the point why FE application written in React was replaced with one written in Angular. It complicates FE integration verification dramatically.

File: backend/cart-service/lambda/src/main.ts (hunk 2 above)
Comment (JenkaBY) [reply backend/cart-service/lambda/src/main.ts]: Good catch! Thanks.

File: backend/cart-service/deploy/src/main/java/by/jenka/rss/backend/cartservice/config/AwsConfig.java (hunk 1 above)
Comment (JenkaBY) [reply backend/cart-service/deploy/src/main/java/by/jenka/rss/backend/cartservice/config/AwsConfig.java:7]: agree

File: backend/cart-service/lambda/src/cart/cart.controller.ts
Comment (JenkaBY) [reply backend/cart-service/lambda/src/cart/cart.controller.ts:1]: Why did I see it only after dead line? Sorry for that. 
IMO all the code is relevant only to this task.

**Pull request merged**